source/miniFastLED.c

```diff
@@ -15,33 +14,33 @@
 led_t rgbArray;
 
 // Convert HSV to RGB and write results to rgbResults
+// Applies current dimming settings.
 void hsv2rgb(uint8_t hue, uint8_t sat, uint8_t val, led_t *rgbResults) {
 
-  // Convert hue, saturation and brightness ( HSV/HSB ) to RGB
-  // "Dimming" is used on saturation and brightness to make
-  // the output more visually linear.
+  // Convert hue, saturation and brightness ( HSV/HSB ) to RGB while applying
+  // dimming to the HSV "Value".
 
   // Apply dimming curves
-  uint8_t value = val;
-  uint8_t saturation = sat;
+  const uint8_t value = dimValue(val);
+  const uint8_t saturation = sat;
 
   // The brightness floor is minimum number that all of
   // R, G, and B will be set to.
-  uint8_t invsat = 255 - saturation;
-  uint8_t brightness_floor = (value * invsat) / 256;
+  const uint8_t invsat = 255 - saturation;
+  const uint8_t brightness_floor = (value * invsat) / 256;
```

Comment:
Original code dims also the brightness floor but that caused white to go green on dimming, so I left it without dimming for now.